index.xml

```diff
@@ -248,27 +248,41 @@ MUST be ignored.
    </section>
 
    <section anchor="canonicalization" title="Signature String Construction">
-
-   <t>
+     <t>
 In order to generate the string that is signed with a key, the client MUST use
 the values of each HTTP header field in the `headers` Signature parameter,
 in the order they appear in the `headers` Signature parameter. It is out of
 scope for this document to dictate what header fields
 an application will want to enforce, but implementers SHOULD at minimum
 include the request target and Date header fields.
-   </t>
-   <t>
-To include the HTTP request target in the signature calculation, use the
-special `(request-target)` header field name.
-    <list style="numbers">
-     <t>
-If the header field name is `(request-target)` then generate the header field
-value by concatenating the lowercased :method, an ASCII space, and
-the :path pseudo-headers (as specified in
-<eref target="https://tools.ietf.org/html/rfc7540#section-8.1.2.3">HTTP/2, Section 8.1.2.3</eref>).
+     </t>
+  
+     <section anchor="sig-pseudo-headers" title="Pseudo-Headers">
+      <t>
+Two special headers are composed for the Signature String that are not
+derived from literal headers in the message. These are the pseudo-headers.
+These special headers encode context for the message,
+specifically the target, host and uri of the request/response, as well as the
+status code and status message provided by the server in a response.
```

Comment:
As stated in the conversation, status message a.k.a reason-phrase should not be included.

"the target, host and uri of the request/response"
The "(request-target)" pseudo-header in it's current form does not include host or the uri. (Uri comprises :scheme, :authority, and :path). And none of these are part of the response.

=>
"These special headers encode context for the message,
specifically the method and path of the request, as well as the
status code provided by the server in a response."

Reply:
> The current "(request-target)" does not include for example the information of ":authority" (i.e. target host+port). Even if this is a breaking change by adding more pseudo-headers, the support for the legacy "(request target)" could still be kept, for better backwards compatibility with the existing implementations.

I've never been very comfortable with the idea of constructing pseudo-headers from already clearly-defined parameters and more so for multi-line and duplicated headers, but as long as they can unambiguously be parsed there's not a particular problem. The issue comes when these are assembled and then can't be broken apart unambiguously - e.g. what if a request target (including authority) has two colons ``:``? Which part was the authority and which was the path? [And so many more...](https://www.youtube.com/watch?v=0uejy9aCNbI)

This may seem trivial for just constructing a string to prove a signature is valid, but in some contexts this could form an actual Electronic Signature or Electronic Seal, with legal implications. The thing being signed suddenly takes on a whack of meaning and consequences.

I would agree that if we're going to create a new pseudo-header, which would be a BC anyway, perhaps time to overhaul the canonicalisation. Since a header name can only be comprised of [tchar](https://tools.ietf.org/html/rfc7230#appendix-B) which excludes colon ``:``, any header starting with a colon is interpreted as a [RFC7540 pseudo-header](https://tools.ietf.org/html/rfc7540#section-8.1.2.1) and not a literal header, though this will lengthen the ``headers`` parameter significantly.